Lib/test/test_httpservers.py

```diff
@@ -539,6 +539,78 @@ def test_get(self):
+        self.assertEqual(response.getheader('content-range'), 'bytes 3-12/30')
+        self.assertEqual(response.getheader('content-length'), '10')
+        self.check_status_and_reason(response, HTTPStatus.PARTIAL_CONTENT, data=self.data[3:13])
+
+        response = self.request(route, headers={'Range': 'bytes=3-'})
+        self.assertEqual(response.getheader('content-range'), 'bytes 3-29/30')
+        self.assertEqual(response.getheader('content-length'), '27')
+        self.check_status_and_reason(response, HTTPStatus.PARTIAL_CONTENT, data=self.data[3:])
+
+        response = self.request(route, headers={'Range': 'bytes=-5'})
+        self.assertEqual(response.getheader('content-range'), 'bytes 25-29/30')
+        self.assertEqual(response.getheader('content-length'), '5')
+        self.check_status_and_reason(response, HTTPStatus.PARTIAL_CONTENT, data=self.data[25:])
+
+        response = self.request(route, headers={'Range': 'bytes=29-29'})
+        self.assertEqual(response.getheader('content-range'), 'bytes 29-29/30')
+        self.assertEqual(response.getheader('content-length'), '1')
+        self.check_status_and_reason(response, HTTPStatus.PARTIAL_CONTENT, data=self.data[29:])
+
+        # end > file size
+        response = self.request(route, headers={'Range': 'bytes=25-100'})
+        self.assertEqual(response.getheader('content-range'), 'bytes 25-29/30')
+        self.assertEqual(response.getheader('content-length'), '5')
+        self.check_status_and_reason(response, HTTPStatus.PARTIAL_CONTENT, data=self.data[25:])
+
+        # invalid ranges
+        response = self.request(route, headers={'Range': 'bytes=100-200'})
+        self.assertEqual(response.getheader('content-range'), 'bytes */30')
+        self.check_status_and_reason(response, HTTPStatus.REQUESTED_RANGE_NOT_SATISFIABLE)
+
+        response = self.request(route, headers={'Range': 'bytes=4-3'})
+        self.check_status_and_reason(response, HTTPStatus.OK, data=self.data)
+
+        response = self.request(route, headers={'Range': 'bytes=wrong format'})
+        self.check_status_and_reason(response, HTTPStatus.OK, data=self.data)
+
+        response = self.request(route, headers={'Range': 'bytes=-'})
+        self.check_status_and_reason(response, HTTPStatus.OK, data=self.data)
+
+        response = self.request(route, headers={'Range': 'bytes=--'})
+        self.check_status_and_reason(response, HTTPStatus.OK, data=self.data)
+
+        response = self.request(route, headers={'Range': 'bytes='})
+        self.check_status_and_reason(response, HTTPStatus.OK, data=self.data)
+
+        # multipart ranges (not supported currently)
+        response = self.request(route, headers={'Range': 'bytes=1-2, 4-7'})
+        self.check_status_and_reason(response, HTTPStatus.OK, data=self.data)
+
+        # empty file
+        with open(os.path.join(self.tempdir_name, 'empty'), 'wb'):
+            pass
```

Comment:
Maybe use https://docs.python.org/3/library/test.html#test.support.os_helper.create_empty_file if `os_helper` is already imported?